### ISSUE TITLE
Add rule for omitting empty enum with associated value arguments

### ISSUE THREAD

**Issue by keith** [realm/SwiftLint#1425]
Currently you can do this:

```swift
enum Foo {
    case bar(String)
}
```

```swift
switch foo {
case .bar(_):
    break
}
```

I think we should add rule to have people remove `(_)`. Note it can also be `(_, _)`.

I've added this regex on our codebase with no false positives: `"\([,\s*_]+\)"`

**Comment by mortenholmgaard** [comment]
I have a method in my enum just like this:
func isCurrent() -> Bool { ... }
And when called like Environment.dev.isCurrent() I get this lint warning in this line which must be wrong.

**Comment by SimplyDanny** [comment]
You should open an own issue for your example with a more detailed description, @mortenholmgaard. Herein, it will probably be missed.

**Comment by mortenholmgaard** [comment]
Ups, I just saw that I didn't have the latest version and this seems to be fixed in that version - sorry.
